packages/flutter/lib/src/material/app.dart

```diff
@@ -234,6 +234,55 @@ class MaterialApp extends StatefulWidget {
   ///
   /// The delegates collectively define all of the localized resources
   /// for this application's [Localizations] widget.
+  ///
+  /// Delegates that produce [WidgetsLocalizations] and [MaterialLocalizations]
+  /// are included automatically. Apps can provide their own versions of these
+  /// localizations by creating implementations of
+  /// [LocalizationsDelegate<WidgetLocalizations>] or
+  /// [LocalizationsDelegate<MaterialLocalizations>] whose load methods return
+  /// custom versions of [WidgetLocalizations] or [MaterialLocalizations].
+  ///
+  /// For example: to add support to [MaterialLocalizations] for a
+  /// locale it doesn't already support, say `const Locale('foo', 'BR')`,
+  /// one could just extend [DefaultMaterialLocalizations]:
+  /// ```
+  /// class FooLocalizations extends DefaultMaterialLocalizations {
+  ///   FooLocalizations(Locale locale) : super(locale);
+  ///   @override
+  ///   String get okButtonLabel {
+  ///     if (locale == const Locale('foo', 'BR'))
+  ///       return 'foo';
+  ///     return super.okButtonLabel;
+  ///   }
```

Comment:
It really seems weird to me that our default localizations delegates don't actually create objects that are locale-specific. Maybe we should remove the "locale" field from these objects to make it more obvious that they are supposed to be constructed in a locale-specific way...

Reply:
The MaterialLocalizations object is locale-specific, it's constructor only saves a reference to the resourceId=>value map for the specified locale. DefaultWidgetsLocalizations initialize its _textDirection field based on the locale.

I was just trying to make this foo-bar example as simple as possible; just to prove that you could get in the way of the default localizations.

Reply:
I worry that people will think this is the way to localize your app, but yeah...

Maybe a better example would be to override something from en-US and have it say something different. Like, if you want to change your `timeOfDayFormat` to something else.